exporter/opentelemetry-exporter-prometheus-remote-write/src/opentelemetry/exporter/prometheus_remote_write/__init__.py

```diff
@@ -0,0 +1,393 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import re
+from collections import defaultdict
+from itertools import chain
+from typing import Dict, Sequence
+
+import requests
+import snappy
+
+from opentelemetry.exporter.prometheus_remote_write.gen.remote_pb2 import (
+    WriteRequest,
+)
+from opentelemetry.exporter.prometheus_remote_write.gen.types_pb2 import (
+    Label,
+    Sample,
+    TimeSeries,
+)
+from opentelemetry.sdk.metrics import Counter
+from opentelemetry.sdk.metrics import Histogram as ClientHistogram
+from opentelemetry.sdk.metrics import (
+    ObservableCounter,
+    ObservableGauge,
+    ObservableUpDownCounter,
+    UpDownCounter,
+)
+from opentelemetry.sdk.metrics.export import (
+    AggregationTemporality,
+    Gauge,
+    Histogram,
+    Metric,
+    MetricExporter,
+    MetricExportResult,
+    MetricsData,
+    Sum,
+)
+
+logger = logging.getLogger(__name__)
+
+PROMETHEUS_NAME_REGEX = re.compile(r"[^\w:]")
+PROMETHEUS_LABEL_REGEX = re.compile(r"[^\w]")
```

Comment:
Was this in the original code or just added? If restored, feel free to tackle in a follow up PR.

The regexes in the [Prometheus docs](https://prometheus.io/docs/concepts/data_model/#:~:text=The%20metric%20name%20specifies%20the%20general%20feature%20of%20a%20system%20that%20is%20measured%20(e.g.%20http_requests_total%20%2D%20the%20total%20number%20of%20HTTP%20requests%20received).%20It%20may%20contain%20ASCII%20letters%20and%20digits%2C%20as%20well%20as%20underscores%20and%20colons.%20It%20must%20match%20the%20regex%20%5Ba%2DzA%2DZ_%3A%5D%5Ba%2DzA%2DZ0%2D9_%3A%5D*.) is different, is that intentional?

Reply:
Ah gotcha I was reading it wrong. Anchoring the \d at the beginning seems to do the trick:

```py
>>> re.sub(r"^\d|[^\w:]", "_", "4hellowor:ld5∂©∑")
'_hellowor:ld5___'
```

lmk if that works and if not feel free to just resolve this comment.

Reply:
[One more requirement](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/data-model.md#prometheus-compatibility:~:text=the%20_%20character.-,Multiple%20consecutive%20_%20characters%20MUST%20be%20replaced%20with%20a%20single%20_%20character.,-The%20Unit%20of). Feel free to open an issue and either do in a follow up PR or I can mark it up for grabs.

Reply:
That link isn't working for me for some reason, but I THINK you're pointing at the consecutive "_" being replaced? I took a whack at that and got it working, but it does clobber the `__name__` label with how things are currently structured. Its fixable, but probably more work than is worth it in this ticket. I will open an issue to avoid bogging this down.

Reply:
Issued opened here:
https://github.com/open-telemetry/opentelemetry-python-contrib/issues/1389